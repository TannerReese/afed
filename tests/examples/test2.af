# Test for circular dependence error

# x Should parse and evaluate correctly
x: {
	y: {
		audio: "good",
		z: y.audio + " " + video.content
			= ``
	},
	video: {
		content: "VIDEO_CONTENT " + x.y.audio
	}
},

# Should produce redefinition error
x: true + true = ``,

# Should produce circular dependence error
video: {
	foo: {x: 5 * bar.y =``},
	bar: {y: baz.z - 6 =``},
	baz: {z: foo.x^2 =``}
}

