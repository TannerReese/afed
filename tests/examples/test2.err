Parse Error in line 17, column 20
	x: true + true = ``,
	                   ^
Error: Redefinition of label in map

1 Parse Error encountered


line 21, column 22 Eval Error: Circular dependence from variable bar.y
line 22, column 22 Eval Error: Circular dependence from variable bar.y
line 23, column 20 Eval Error: Circular dependence from variable bar.y
line 27, column 12 Eval Error: Unary operator - not implemented for type string
line 28, column 13 Eval Error: Binary operator * not implemented between types number and boolean
line 29, column 20 Eval Error: Binary operator + not implemented between types string and array
6 Eval Errors encountered
