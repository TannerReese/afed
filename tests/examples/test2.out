# Test for circular dependence error
# Test for type errors

# x Should parse and evaluate correctly
x: {
	y: {
		audio: "good",
		z: y.audio + " " + video.content
			= `"good VIDEO_CONTENT good"`
	},
	video: {
		content: "VIDEO_CONTENT " + x.y.audio
	}
},

# Should produce redefinition error
x: true + true = `false`,

# Should produce circular dependence error
video: {
	foo: {x: 5 * bar.y =`Eval Error: Circular dependence from variable bar.y`},
	bar: {y: baz.z - 6 =`Eval Error: Circular dependence from variable bar.y`},
	baz: {z: foo.x^2 =`Eval Error: Circular dependence from variable bar.y`}
},

# Should produce type errors
-"Hello" = `Eval Error: Unary operator - not implemented for type string`,
1 * false = `Eval Error: Binary operator * not implemented between types number and boolean`,
"Hello" + [1, 2] = `Eval Error: Binary operator + not implemented between types string and array`,

